obspy/io/xseed/parser.py

```diff
@@ -670,6 +671,155 @@ def write_resp(self, folder, zipped=False):
+        # Make unit lookup blockette34
+        b34s = ('034  44  4M/S~velocity in meters per second~',
+                '034  25  5V~emf in volts~',
+                '034  32  7COUNTS~digital counts~'
+                )
+        abbv_lookup = {'M/S': '4', 'V': '5', 'COUNTS': '7'}
+
+        for b34 in b34s:
+            data = io.BytesIO(b34.encode('utf-8'))
+            b34_obj = blockette.Blockette034(debug=self.debug,
+                                             record_type='A')
+            b34_obj.parse_seed(data, expected_length=len(b34))
+            self.temp['abbreviations'].append(b34_obj)
+        self.temp['stations'].append([])
+        root_attribute = self.temp['stations'][-1]
+
+        for RESPblockettefieldlist in blockettelist:
+            # Create a new blockette using the first field
+            RESPblockette_id, RESPfield, resp_value = RESPblockettefieldlist[0]
+            class_name = 'Blockette%03d' % int(RESPblockette_id)
+            blockette_class = getattr(blockette, class_name)
+            record_type = record_type_from_blocketteid(blockette_class.id)
+            blockette_obj = blockette_class(debug=self.debug,
+                                            strict=False,
+                                            compact=False,
+                                            record_type=record_type)
+            blockette_fields = (blockette_obj.default_fields +
+                                blockette_obj.get_fields())
+            unrolled_blockette_fields = list()
+            for bf in blockette_fields:
+                if isinstance(bf, Loop):
+                    for df in bf.data_fields:
+                        unrolled_blockette_fields.append(df)
+                else:
+                    unrolled_blockette_fields.append(bf)
+            blockette_fields = unrolled_blockette_fields
+            # List of fields with fields used removed,
+            # so unused can be set to default after
+            unused_fields = blockette_fields[:]
+
+            for RESPblockette_id, RESPfield, resp_value in RESPblockettefieldlist:
+                for bfield in blockette_fields:
+                    if bfield.id == int(RESPfield):
+                        if isinstance(bfield, VariableString):
+                            # Variable string needs terminator '~'
+                            resp_value += '~'
+                        # print(RESPvalue)
+                        # Lookup if abbv
+                        resp_value = abbv_lookup.get(resp_value, resp_value)
+                        data_resp_value = io.BytesIO(resp_value.encode('utf-8'))
+                        if (hasattr(bfield, 'length') and
+                                bfield.length < len(resp_value)):
+                            # RESP does not use the same length for floats
+                            # as SEED does
+                            bfield.length = len(resp_value)
```

Comment:
bfield is an instance, BUT is defined in a class variable of blockette. Changing this changes the class and all instances of it.